src/fs_storage.c

```diff
@@ -0,0 +1,77 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2012-2018 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <sys/statvfs.h>
+
+#include "swupd.h"
+
+/* recursively calculate size from a complete list of bundle_result structs */
+long calculate_size(char *bname, struct list *bundle_info, bool installed)
+{
+	struct list *ptr;
+	struct bundle_result *bi;
+	long size = 0;
+	ptr = list_head(bundle_info);
+	while (ptr) {
+		bi = ptr->data;
+		ptr = ptr->next;
+		if (bi->seen || strcmp(bname, bi->bundle_name) != 0) {
+			continue;
+		}
+
+		/* If installed is true the initial call to this recursive
+		 * function was for an installed bundle. In this case we want
+		 * to calculate the contentsize of all included bundles (we
+		 * know they are all installed already) so we can report the
+		 * total installed size on the system.
+		 *
+		 * Otherwise only add the contentsize of bundles not already
```

Comment:
This comment worries me, What happens if a bundle grows a lot in size, e.g. editors gets neovim added as another editor?

I know that this is just "moved" code and the comment is already present, but has the calling circumstances changed too much for it still to be valid?

Reply:
Well since this is only used in the bundle-add codepath the size of bundles already on the system shouldn't change since it isn't updating them.